### PR TITLE
pass ctx objects from cmd Run func down to the kubeClients

### PR DESCRIPTION
**Pull request by tariq1890**
Fixing how the ctx object is initalised and passed down the call stack in the demo code

This PR fixes the following linting issues

```
golangci-lint run ./...
cmd/dra-example-controller/driver.go:112:19: Function `Get` should pass the context parameter (contextcheck)
        err := client.Get()
                         ^
cmd/dra-example-controller/driver.go:142:21: Function `Update` should pass the context parameter (contextcheck)
        err = client.Update(&crd.Spec)
                           ^
cmd/dra-example-controller/driver.go:168:19: Function `Get` should pass the context parameter (contextcheck)
        err := client.Get()
                         ^
cmd/dra-example-controller/driver.go:194:21: Function `Update` should pass the context parameter (contextcheck)
        err = client.Update(&crd.Spec)
                           ^
cmd/dra-example-controller/driver.go:228:19: Function `Get` should pass the context parameter (contextcheck)
        err := client.Get()
                         ^
```

### CODE REVIEW

File: cmd/dra-example-kubeletplugin/cdi.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import (
 
 	cdiapi "github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
 	cdispec "github.com/container-orchestrated-devices/container-device-interface/specs-go"
+
```
Comment (pohly) [inline cmd/dra-example-kubeletplugin/cdi.go:25]: This is unrelated to the PR, but I agree that this is better, so let's leave this in.

File: pkg/example.com/resource/clientset/versioned/clientset.go [hunk 2]
```diff
@@ -22,11 +22,11 @@ import (
 	"fmt"
 	"net/http"
 
-	gpuv1alpha1 "sigs.k8s.io/dra-example-driver/pkg/example.com/resource/clientset/versioned/typed/gpu/v1alpha1"
-	nasv1alpha1 "sigs.k8s.io/dra-example-driver/pkg/example.com/resource/clientset/versioned/typed/nas/v1alpha1"
 	discovery "k8s.io/client-go/discovery"
 	rest "k8s.io/client-go/rest"
 	flowcontrol "k8s.io/client-go/util/flowcontrol"
+	gpuv1alpha1 "sigs.k8s.io/dra-example-driver/pkg/example.com/resource/clientset/versioned/typed/gpu/v1alpha1"
```
Comment (pohly) [inline pkg/example.com/resource/clientset/versioned/clientset.go:28]: Should there be a blank line above this to separate third-party imports from our own code?

**Comment by tariq1890** [comment]
@pohly I reverted those changes. THanksq for reviewing!

File: cmd/dra-example-kubeletplugin/cdi.go (hunk 1 above)
Comment (tariq1890) [reply cmd/dra-example-kubeletplugin/cdi.go:25]: Thanks!

File: .golangci.yaml [hunk 3]
```diff
@@ -13,3 +14,7 @@ linters:
 run:
   tests: true
   timeout: 1m
+
+linters-settings:
+  goimports:
+    local-prefixes: "sigs.k8s.io/dra-example-driver"
```
Comment (pohly) [inline .golangci.yaml:20]: This also should go into a separate PR which cleans up the import grouping.

**Review by pohly — CHANGES_REQUESTED** [review]
Looks good to me, but needs a rebase and one nit.

File: .golangci.yaml (hunk 3 above)
Comment (tariq1890) [reply .golangci.yaml]: Done, I removed that change

**Review by pohly — APPROVED** [review]
/lgtm
/approve

Thanks! Your other changes from an earlier revision of this PR also looked good, please submit in a new PR.
